### DISCUSSION TITLE
Codeblock

### DISCUSSION THREAD

**Discussion by NeonOdyssey** [austintoddj/canvas#985] [New Features]
When using the code block I have noticed that the formatting is set in stone.
I feel like we should be able to set what language the code block formatting should display.
Just like how @PrismJS does things or even markdown where you would do the 3 ``` then set the language on the same line ...

```markdown
---
format: this is markdown formatting
---
**Bold** and *italic* with some `code`
```
and 
```php
class codeblock {
  $format = "this is php formatting"; 

  public function () => {
    // Prints "this is php formatting"
    echo $this->format();
  }
}
```

**Comment by austintoddj** [comment]
Since I'm really intent in making [TipTap](https://www.tiptap.dev/) the default editor in the next version, I'll see if this is possible with their API.

**Comment by austintoddj** [comment]
Closing this out with [Canvas v7.0.0](https://github.com/austintoddj/canvas/releases/tag/v7.0.0).

The TipTap editor supports code blocks with a **language** selector (syntax highlighting via lowlight) — the capability called for here.
